ftw/solr/converters.py

```diff
@@ -1,13 +1,16 @@
 from DateTime import DateTime
 from datetime import datetime
 from datetime import date
+import math
 
 
 def to_iso8601(value, multivalued=False):
     if isinstance(value, DateTime):
         v = value.toZone('UTC')
-        value = u'%04d-%02d-%02dT%02d:%02d:%06.3fZ' % (
-            v.year(), v.month(), v.day(), v.hour(), v.minute(), v.second()
+        millisecond, second = math.modf(v.second())
+        value = u'%04d-%02d-%02dT%02d:%02d:%02d.%03dZ' % (
+            v.year(), v.month(), v.day(), v.hour(), v.minute(),
+            int(second), millisecond * 1000
```

Comment:
Works for me, though I'd find `int(v.second()*1000)/1000` or `math.floor(v.second()*1000)/1000` almost easier to read.